prog.go

```diff
@@ -745,6 +745,8 @@ type RunOptions struct {
 	// CPU to run Program on. Optional field.
 	// Note not all program types support this field.
 	CPU uint32
+	// BatchSize is the number of executions per syscall.
```

Comment:
I had to look up what this is actually for, and I think the niche-ness of it warrants some elaboration. If I understand the manual correctly, this doesn't affect the amount of syscalls made by the library (that's what the `Repeat` option does).

```suggestion
	// BatchSize (default 64) affects the kernel's packet buffer allocation behaviour when running
	// programs with BPF_F_TEST_XDP_LIVE_FRAMES and a non-zero [RunOptions.Repeat] value.
	// For more details, see the kernel documentation on BPF_PROG_RUN:
	// https://docs.kernel.org/bpf/bpf_prog_run.html#running-xdp-programs-in-live-frame-mode
```

I'm not sure if it makes sense to test this flag explicitly, we'd mostly be exercising the kernel itself. Getting a new entry for this in `examples/` would be really nice, though!

Reply:
> If I understand the manual correctly, this doesn't affect the amount of syscalls made by the library (that's what the Repeat option does).

So `Repeat` also does not change the amount of syscalls made. `Repeat` does what it sounds like, for every syscall, the kernel will invoke the program `Repeat` amount of times.

The `BatchSize` is specific to the packet generation with XDP scenario. Its the batch size of the amount of packets to push down to the network stack at once. So lets say you set `Repeat` to 10.000, and `BatchSize` to 64. You will get 1 syscall, it will execute the XDP program 10.000 times, and it will "flush" the resulting packets every 64 invocations to the network stack. Currently `BatchSize` can go up to 256, which gives a performance boost over the default of 64.